### ISSUE TITLE
Removed Posts should not be visible for non-moderator users

### ISSUE THREAD

**Issue by odlbot** [mitodl/open-discussions#2785]
### Steps to Reproduce
Ideally, when items are removed, they should no longer be visible for anonymous and non-moderator users. Currently, while no longer in the channel and front pages, removed items can still be accessed directly from the URL. 

Steps:

1. Go to URL of item to remove. Copy the URL.
1. Remove the item.
1. Log out of Open or open another browser that isn't logged in.
1. Paste URL of removed item and access that page.
1. You can view the removed post.

### Expected Behavior
Ideally there'd be a 404 for non-moderator and anonymous users when they try to view removed items.

### Actual Behavior
Everyone can still view the removed items directly with the URL, even after removed.

**Comment by aliceriot** [comment]
just started looking into this, there's a bit of an issue. currently in our serializer for post we set `is_removed` to `True` when the `post.banned_by` field is populated. Basically, after a moderator removes a post this field will be populated with their username.

BUT! the problem is that, as far as I can tell, this field is *only populated when moderators are accessing the resource*. For non-moderator accounts and for anonymous accounts the `banned_by` field is always empty, regardless of whether the post has been removed or not.

And there doesn't seem to be a reliable way other than that of determining whether a post has been removed or not. If the post is a text post the text will be set to `[removed]`, but this is not the case for link posts.

So that is pretty unfortunate I think. I don't see a great way to resolve this issue, because we'd need to do something like

- add a "removed" field to our `Post` model
- start saving that value whenever a post is removed or approved
- design a backfill script that would be run under the indexing user account that could look at *every single post* and determine whether it was removed or not.

we'll have to have that in place before we can implement a 404 page on the frontend for removed posts. Once we have an accurate `.removed` field from the post API it's a 2-3 line change on the frontend to make this work, but unfortunately it seems we have a faulty assumption baked in to our API (namely that `banned_by` will be populated for all users).

@rhysyngsun @noisecapella I'm wondering if either of you did work with this and may know a different, possibly indirect, way to tell if a post has been removed?
